community/kernel/src/main/java/org/neo4j/kernel/impl/api/index/BatchingMultipleIndexPopulator.java

```diff
@@ -334,6 +344,8 @@ private static String allStackTraces()
      */
     private class BatchingIndexPopulation extends IndexPopulation
     {
+        boolean failed;
```

Comment:
`BatchingIndexPopulation` is not accessed by multiple threads. Only thread that performs store scan calls `#addApplicable()` and flips population to failed.